internals-js/src/operations.ts

```diff
@@ -995,6 +1017,72 @@ export class NamedFragments {
     return mapped;
   }
 
+  /**
+   * This method modifies the fragment by applying the provided mapper to the selection set of the fragments,
+   * but it does so after expanding all the nested fragments, and re-fragment after application.
+   * This method also allows skipping some of the fragments by having the matter return undefined.
+   */
+  mapToExpandedSelectionSets(
+    mapper: (selectionSet: SelectionSet) => SelectionSet | undefined,
+    recreateFct: (frag: NamedFragmentDefinition, newSelectionSet: SelectionSet) => NamedFragmentDefinition = (f, s) => f.withUpdatedSelectionSet(s),
+  ): NamedFragments | undefined {
+    type FragmentInfo = {
+      original: NamedFragmentDefinition,
+      expandedSelectionSet: SelectionSet,
+      dependentsOn: string[],
+    };
+    const fragmentsMap = new Map<string, FragmentInfo>();
+
+    const removedFragments = new Set<string>();
+    for (const fragment of this.definitions()) {
+      const expandedSelectionSet = mapper(fragment.selectionSet.expandAllFragments().trimUnsatisfiableBranches(fragment.typeCondition));
+      if (!expandedSelectionSet) {
+        removedFragments.add(fragment.name);
+        continue;
+      }
+
+      const otherFragmentsUsages = new Map<string, number>();
+      fragment.collectUsedFragmentNames(otherFragmentsUsages);
+      fragmentsMap.set(fragment.name, {
+        original: fragment,
+        expandedSelectionSet,
+        dependentsOn: Array.from(otherFragmentsUsages.keys()),
```

Comment:
I think this should be `dependsOn` or `dependencies`, a "dependent" is the inverse so the naming is confusing here.